eth/stagedsync/stage_log_index.go

```diff
@@ -513,9 +508,7 @@ func pruneLogIndex(logPrefix string, tx kv.RwTx, tmpDir string, pruneFrom, prune
 						return err
 					}
 				}
-				if err := pruneLogKeyCollector.Collect(k, nil); err != nil {
-					return err
-				}
+				c.DeleteCurrent()
```

Comment:
1. err not checked 
2. plz use `tx.Delete(table, k)`. Because seems some version of mdbx has bug when same cursor used for `for` iteration and `DeleteCurrent` - we did workaround it by using 2 cursors or `tx.Delete`, tnx. I'm not sure which versions are affected - because not easy to reproduce.

Reply:
- `tx.DeleteCurrent` pattern is used at many places in the code base. Are you also saying all of them need fixing too, then? Because the way i see it, `DeleteCurrent` can only be used in a for loop like that.
- What is the performance impact of using `tx.Delete` here, compared to `DeleteCurrent`?

Reply:
DeleteCurrent is not bad - use 1 cursor to iterate and in same time DeleteCurrent - bad in some mdbx versions. 

No performance impact (of course it’s there - O(1) vs O(n log n), but mdbx’s methods itself are fast-enough, bottleneck is usually in “touching cold data”, but you touching it by loop anyway).

Don’t need modify other places.

Reply:
Okay, for now, using `tx.Delete` in its place